server/lib/nl_detection.py

```diff
@@ -0,0 +1,65 @@
+# The following classes are used for the NL Detection.
+from abc import ABC
+from dataclasses import dataclass
+from typing import Dict, List
+
+
+@dataclass
+class Place:
+  """Place attributes."""
+  dcid: str
+  name: str
+
+
+@dataclass
+class SV:
+  """Statistical Variable attributes."""
+  dcid: str
+  name: str
+  place_type: str
+
+
+@dataclass
+class PlaceDetection:
+  """Various attributes of place detection."""
+  query: str
+  query_without_place: str
+  places_found: List[str]
+  main_place: Place
+
+
+@dataclass
+class SVDetection:
+  """Various attributes of SV detection."""
+  query: str
+  sv2scores: Dict[str, float]
+
+  highlight_svs: List[str]
+  relevant_svs: List[str]
```

Comment:
is relevant SVs from the model or from the hierarchy? If from hierarchy, maybe not including here?

Reply:
Yeah, good point. I thought you'd want that here since it was filtering based on scores. But it is something the middle-ware logic can decide to update/change and set other thresholds too.